client/components/overlay/overlay.jsx

```diff
@@ -100,7 +100,7 @@ module.exports = React.createClass({
 				</TitleData>
 
 				<div className="wp-content" ref="overlayInnerContent">
-					<NoticesList id="overlay-notices" notices={ notices.list }/>
+					<GlobalNotices id="overlay-notices" notices={ notices.list }/>
```

Comment:
Also, one more thing: `Overlay` isn't part of `Layout` in this case, so `GlobalNotices` didn't have access to the global state here. This was fixed in https://github.com/Automattic/wp-calypso/pull/1944